impl.go

```diff
@@ -83,18 +83,31 @@ func findInterface(iface string, srcDir string) (path string, id string, err err
 	if err != nil {
 		panic(err)
 	}
-	if len(f.Imports) == 0 {
+
+	qualified := strings.ContainsRune(iface, '.')
+
+	if len(f.Imports) == 0 && qualified {
 		return "", "", fmt.Errorf("unrecognized interface: %s", iface)
 	}
-	raw := f.Imports[0].Path.Value   // "io"
-	path, err = strconv.Unquote(raw) // io
-	if err != nil {
-		panic(err)
+
+	declIdx := 0
```

Comment:
A comment here would be helpful. Something like:

```go
// If qualified, the code looks like:
//
// package hack
//
// import (
//   "io"
// )
//
// var i io.Reader
//
// If !qualified, the code looks like:
//
// package hack
//
// var i Reader
```

There's not much shared code here. I suspect that this may be more readable fully separated.

```go
if qualified {
  // code as it was, including return statement
}
// new code that handles !qualified case
```